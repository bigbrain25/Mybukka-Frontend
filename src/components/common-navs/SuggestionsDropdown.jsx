import React from 'react';
import PropTypes from 'prop-types';

import UseCurrentLocation from './UseCurrentLocation';
import GeoSuggestions from './GeoSuggestions';

const SuggestionsDropdown = ({ handleClick }) => (
  <div className="suggestion-dropdown">
    <UseCurrentLocation
      handleClick={handleClick}
    />
    <GeoSuggestions
      handleClick={handleClick}
      suggestions={[
        { location: 'Mende Maryland, Lagos', key: '1' },
        { location: 'Mende Maryland, Lagos', key: '2' },
        { location: 'Mende Maryland, Lagos', key: '3' },
      ]}
    />
  </div>
);

export default SuggestionsDropdown;

SuggestionsDropdown.defaultProps = {
  handleClick: () => {},
};

SuggestionsDropdown.propTypes = {
  handleClick: PropTypes.func,
};
