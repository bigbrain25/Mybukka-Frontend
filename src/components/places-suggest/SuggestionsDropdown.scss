.suggestion-dropdown {
  z-index: 30;
  position: relative;
}
