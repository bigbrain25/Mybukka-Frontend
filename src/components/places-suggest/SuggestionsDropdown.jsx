import React from 'react';
import PropTypes from 'prop-types';

import UseCurrentLocation from './UseCurrentLocation';
import GeoSuggestions from './GeoSuggestions';

import './SuggestionsDropdown.scss';

const SuggestionsDropdown = ({
  setLocation,
  predictions,
  useCurrentLocationVisible,
}) => (
  <div className="suggestion-dropdown">
    {useCurrentLocationVisible &&
      <UseCurrentLocation />
    }
    <GeoSuggestions
      handleClick={setLocation}
      predictions={predictions}
    />
  </div>
);

export default SuggestionsDropdown;

SuggestionsDropdown.defaultProps = {
  useCurrentLocationVisible: true,
};

SuggestionsDropdown.propTypes = {
  setLocation: PropTypes.func.isRequired,
  useCurrentLocationVisible: PropTypes.bool
};
