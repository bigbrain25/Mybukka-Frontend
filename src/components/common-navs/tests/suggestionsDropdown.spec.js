import React from 'react';

import SuggestionsDropdown from '../SuggestionsDropdown';

describe('Suggestions dropdown component', () => {
  const props = {
    handleClick: jest.fn(),
  };
  const wrapper = shallow(<SuggestionsDropdown {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
