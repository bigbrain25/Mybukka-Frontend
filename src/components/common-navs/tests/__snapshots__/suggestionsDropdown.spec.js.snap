// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suggestions dropdown component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SuggestionsDropdown
    handleClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <UseCurrentLocation
          handleClick={[MockFunction]}
        />,
        <GeoSuggestions
          handleClick={[MockFunction]}
          suggestions={
            Array [
              Object {
                "key": "1",
                "location": "Mende Maryland, Lagos",
              },
              Object {
                "key": "2",
                "location": "Mende Maryland, Lagos",
              },
              Object {
                "key": "3",
                "location": "Mende Maryland, Lagos",
              },
            ]
          }
        />,
      ],
      "className": "suggestion-dropdown",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleClick": [MockFunction],
          "suggestions": Array [
            Object {
              "key": "1",
              "location": "Mende Maryland, Lagos",
            },
            Object {
              "key": "2",
              "location": "Mende Maryland, Lagos",
            },
            Object {
              "key": "3",
              "location": "Mende Maryland, Lagos",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <UseCurrentLocation
            handleClick={[MockFunction]}
          />,
          <GeoSuggestions
            handleClick={[MockFunction]}
            suggestions={
              Array [
                Object {
                  "key": "1",
                  "location": "Mende Maryland, Lagos",
                },
                Object {
                  "key": "2",
                  "location": "Mende Maryland, Lagos",
                },
                Object {
                  "key": "3",
                  "location": "Mende Maryland, Lagos",
                },
              ]
            }
          />,
        ],
        "className": "suggestion-dropdown",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleClick": [MockFunction],
            "suggestions": Array [
              Object {
                "key": "1",
                "location": "Mende Maryland, Lagos",
              },
              Object {
                "key": "2",
                "location": "Mende Maryland, Lagos",
              },
              Object {
                "key": "3",
                "location": "Mende Maryland, Lagos",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
